### ISSUE TITLE
False positive 'super-with-arguments' if there are kwargs

### ISSUE THREAD

**Issue by skirpichev** [PyCQA/pylint#5768]
### Bug description

```
$ cat -n a.py 
     1  class A:
     2      def __init__(self, a=0):
     3          self.a = a
     4  
     5  
     6  class B(A):
     7      def __init__(self, a=0):
     8          super(B, self).__init__(a)
     9          self.a += 1
```

### Configuration

```ini
none
```

### Command used

```shell
pylint -d all -e super-with-arguments a.py
```

### Pylint output

```shell
************* Module a
a.py:8:8: R1725: Consider using Python 3 style super() without arguments (super-with-arguments)

------------------------------------------------------------------
Your code has been rated at 8.57/10 (previous run: 8.57/10, +0.00)
```

### Expected behavior

No error

### Pylint version

```shell
pylint 2.12.2
astroid 2.9.3
Python 3.10.0+ (heads/3.10:045f205ba4, Nov  8 2021, 05:21:54) [GCC 10.2.1 20210110]
```

**Comment by DanielNoord** [comment]
```console
❯ cat test.py
class A:
    def __init__(self, a=0):
        self.a = a


class B:
    def __init__(self, a=0):
        super(B, self).__init__(a)
        self.a += 1


B()
❯ python test.py
Traceback (most recent call last):
  File "/Users/daniel/Documents/Geschiedenis/KNIR/test.py", line 12, in <module>
    B()
  File "/Users/daniel/Documents/Geschiedenis/KNIR/test.py", line 8, in __init__
    super(B, self).__init__(a)
TypeError: object.__init__() takes exactly one argument (the instance to initialize)
```

The code from your example crashes. Am I missing something here?

**Comment by skirpichev** [comment]
Thanks, example fixed.

**Comment by DanielNoord** [comment]
```console
❯ cat test.py
class A:
    def __init__(self, a=0):
        self.a = a


class B(A):
    def __init__(self, a=0):
        super(B, self).__init__(a)
        self.a += 1


class C(A):
    def __init__(self, a=0):
        super().__init__(a)
        self.a += 1


print(B().a)
print(B(2).a)
print(C().a)
print(C(2).a)
❯ python test.py
1
3
1
3
```

What's the difference between B and C? Isn't this message correct here?

**Comment by skirpichev** [comment]
> What's the difference between B and C?

None, from the language point of view.  But pylint apparently thinks there is...

>  Isn't this message correct here?

No.  There is no error for L14 (super call in the `C.__init__`).  Same should be for the B class.

**Comment by DanielNoord** [comment]
I think the message is referring to the fact that you can write `B` as `C`. There is no need for the arguments in `super(...)` in Python 3 and thus `super(B, self).__init__(a)` can be simplified to ` super().__init__(a)`. That message seems correct in this case?

**Comment by skirpichev** [comment]
Thanks you are right, sorry for bothering.

**Comment by DanielNoord** [comment]
No worries! I can see how the message can be confusing at first!
